### ISSUE TITLE
[OpenAPI] Not adding securitySchemas from document transformers

### ISSUE THREAD

**Issue by VeithBuergerhoff** [dotnet/aspnetcore#59454]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

Flows added to the security schema using document transformers are not being rendered into the openapi json file.

Creating a new webapi project and modifying the call to `AddOpenApi` to include a relevant document transformer like this:

```csharp
builder.Services.AddOpenApi(options => options.AddDocumentTransformer((document, _, _) =>
{
    var requirements = new Dictionary<string, OpenApiSecurityScheme>
    {
        ["Bearer"] = new OpenApiSecurityScheme
        {
            Flows = new OpenApiOAuthFlows
            {
                Password = new OpenApiOAuthFlow
                {
                    TokenUrl = new Uri("https://localhost"),
                    Scopes = new Dictionary<string, string>
                    {
                        ["api"] = "API access"
                    }
                }
            },
            Name = IdentityConstants.BearerScheme,
            Type = SecuritySchemeType.Http,
            Scheme = "Bearer",
        }
    };
    document.Components ??= new OpenApiComponents();
    document.Components.SecuritySchemes = requirements;
    return Task.CompletedTask;
}));
```

Adds the following securitySchema:

```json
"securitySchemes": {
    "Bearer": {
        "type": "http",
        "scheme": "Bearer"
    }
}
```

### Expected Behavior

The json file includes the flow in the securitySchema:

```json
"securitySchemes": {
    "Bearer": {
        "flows": {
            "password": {
                "tokenUrl": "https://localhost",
                "scopes": {
                    "api": "API access"
                }
            }
        },
        "type": "http",
        "scheme": "Bearer"
    }
}
```

### Steps To Reproduce

Check out [this](https://github.com/VeithBuergerhoff/dotnet-openapi-missing-flows) repository.

### .NET Version

9.0.100

**Comment by martincostello** [comment]
It's a bit counter-intuitive from the Microsoft.OpenApi library, but you typically need to also register a the security requirement too for things to work. [For example](https://github.com/martincostello/aspnetcore-openapi/blob/b83950a73cf15a142d186d6b5479540db099973c/src/TodoApp/OpenApi/AspNetCore/AspNetCoreOpenApiEndpoints.cs#L35-L53):

```diff
var scheme = new OpenApiSecurityScheme()
{
    BearerFormat = "JSON Web Token",
    Description = "Bearer authentication using a JWT.",
    Scheme = "bearer",
    Type = SecuritySchemeType.Http,
    Reference = new()
    {
        Id = "Bearer",
        Type = ReferenceType.SecurityScheme
    }
};


document.Components ??= new();
document.Components.SecuritySchemes ??= new Dictionary<string, OpenApiSecurityScheme>();
document.Components.SecuritySchemes[scheme.Reference.Id] = scheme;

+ // Also register the scheme with the security requirements
+ document.SecurityRequirements ??= [];
+ document.SecurityRequirements.Add(new() { [scheme] = [] });
```

The same is true [with Swashbuckle.AspNetCore](https://github.com/martincostello/aspnetcore-openapi/blob/b83950a73cf15a142d186d6b5479540db099973c/src/TodoApp/OpenApi/Swashbuckle/SwashbuckleOpenApiEndpoints.cs#L36-L50) which uses the same underlying library:

```diff
var scheme = new OpenApiSecurityScheme()
{
    BearerFormat = "JSON Web Token",
    Description = "Bearer authentication using a JWT.",
    Scheme = "bearer",
    Type = SecuritySchemeType.Http,
    Reference = new()
    {
        Id = "Bearer",
        Type = ReferenceType.SecurityScheme
    },
    UnresolvedReference = false
};

options.AddSecurityDefinition(scheme.Reference.Id, scheme);

+ // Also register the scheme with the security requirements
+ options.AddSecurityRequirement(new() { [scheme] = [] });
```

**Comment by VeithBuergerhoff** [comment]
Thanks for the quick reply!
Unfortunately, this does not fix the issue.
I did not specify the security requirement, because it does not impact the security schema, since it is already faulty.

The security requirement is, well, required to apply the security schema globally or to select operations.
Adding a requirement on the document adds the following section:

```json
  "security": [
    {
      "Bearer": [ ]
    }
  ]
```

I did a sanity check and applied it to the linked repo, but the flows are still missing in the generated document.
